### ISSUE TITLE
Menu is disappearing with Windows light theme

### ISSUE THREAD

**Issue by JeromeMartinez** [MediaArea/MediaInfo#898]
Action to reproduce with latest main branch:
- Select light theme
- drag and drop a file

![image](https://github.com/MediaArea/MediaInfo/assets/1641638/02eb4253-2232-4ade-9eb3-9fbd2ba4029a)

The menu items disappear until there is a mouse over the items, as if there is a lack of refresh somewhere.
It is totally fine with dark theme.

@cjee21 this is an example of why we are reluctant to change things, there are always some regressions to fix :(.
But we still want the changes, just that we need to find out which one produced that.

**Comment by cjee21** [comment]
This does not happen previously? 🤔 How does this happen when there are no changes to the menu bar...

**Comment by cjee21** [comment]
> Action to reproduce with latest main branch:
> 
>     * Select light theme
> 
>     * drag and drop a file
> 

Cannot reproduce on both 24.06 and self-build from local branch (current main+64-bit+WebView2 merged).


https://github.com/MediaArea/MediaInfo/assets/77721854/0eca9b5b-5280-4d0e-811d-eca42a4f7ef0

**Comment by JeromeMartinez** [comment]
On my machine:
[20240627-2](https://mediaarea.net/download/snapshots/binary/mediainfo-gui/20240627-2/MediaInfo_GUI_24.05.20240627_Windows.exe) OK
[20240628-2](https://mediaarea.net/download/snapshots/binary/mediainfo-gui/20240628-2/MediaInfo_GUI_24.06.20240628_Windows.exe) there is a start of not showing the bottom of the menu during drag and drop
[20240630-2](https://mediaarea.net/download/snapshots/binary/mediainfo-gui/20240630-2/MediaInfo_GUI_24.06.20240630_Windows.exe) menu disappears during drag and drop

(20240628-2: 
![image](https://github.com/MediaArea/MediaInfo/assets/1641638/52384965-ffd7-4e72-956c-59504bbb8cf7))

So even if it is only on my machine, there is really a bad something which changed somewhere in the last days. After the release.
Could you check the 2 last links on your machine?

**Comment by cjee21** [comment]
I have an idea. What is move tabs to bottom then push it out of the window boundary. Since nothing is at the bottom, nothing should go wrong?

**Comment by JeromeMartinez** [comment]
>  Since nothing is at the bottom, nothing should go wrong?

There is something at the bottom. Try with:
[Root_extra.zip](https://github.com/user-attachments/files/16050901/Root_extra.zip)

> That is the attempt to hide the tabs (again).

OK, so it is linked to the other things in another ticket, when the menu shows additional items.
So I think it is a need of refactor of the menu related algorithm, I am sure there is something to do for reducing the count of calls of such code which should remove the glitches.

I guess it was hacks on hacks with collateral damage hidden during years and now that there is a clean up of some other parts of the code it becomes visible :(.
@cjee21 if you are motivated I let you do some refactoring of this part of the code, else I try to do it this evening.

**Comment by cjee21** [comment]
> > Since nothing is at the bottom, nothing should go wrong?
> 
> There is something at the bottom. Try with: [Root_extra.zip](https://github.com/user-attachments/files/16050901/Root_extra.zip)

How am I supposed to try with this? What should I do with the XML?

**Comment by cjee21** [comment]
Alright. This issue is fixed perfectly on all DPIs.

But I discovered that the button does not look good at 200%.
![Screenshot 2024-07-01 175402](https://github.com/MediaArea/MediaInfo/assets/77721854/112d7b83-ee57-446e-a1e6-5232ce1858e9)

**Comment by JeromeMartinez** [comment]
> But I discovered that the button does not look good at 200%.

The button at the bottom is not supposed to be so big!

**Comment by cjee21** [comment]
> > But I discovered that the button does not look good at 200%.
> 
> The button at the bottom is not supposed to be so big!

Any idea where went wrong and why it is not scaling in the same way as other elements?

**Comment by JeromeMartinez** [comment]
> Alright. This issue is fixed perfectly on all DPIs.

Curious about how you fix it, waiting for the PR ☺️.

**Comment by JeromeMartinez** [comment]
> Any idea where went wrong and why it is not scaling in the same way as other elements?

No, IIRC it is just a button. The only one we have IIRC so not possible to compare.

**Comment by cjee21** [comment]
> No, IIRC it is just a button. The only one we have IIRC so not possible to compare.

The button next to the filename selector and the many buttons in Preferences window are different?

**Comment by JeromeMartinez** [comment]
> The button next to the filename selector and the many buttons in Preferences window are different?

They are not enlarged for following the windows size. Else I guess they are same.

**Comment by cjee21** [comment]
> Curious about how you fix it, waiting for the PR ☺️.

To sum up, the cause of this menu bar graphics glitch issue exists (partially missing) since at least 17 years ago due to
https://github.com/MediaArea/MediaInfo/blob/10705513993a835078ec8927aaa8e40eccb8bcdf/Source/GUI/VCL/GUI_Main.cpp#L162
and is made worse (all missing) by the recent change to
https://github.com/MediaArea/MediaInfo/blob/e918631c5f99eff9cf11fc9b89c8a9e3bbff69ba/Source/GUI/VCL/GUI_Main.cpp#L248
and will now be perfectly fixed by the linked PR.

I have scrutinized the borders at 200% DPI and they look pixel-perfect 😄

**Comment by JeromeMartinez** [comment]
> and will now be perfectly fixed by the linked PR.

You are starting to know the GUI code better than its author (who forgot everything about it 😅, it was so a long time ago).

**Comment by cjee21** [comment]
> The button at the bottom is not supposed to be so big!

Fixed. Also fixed the issue of buttons next to drop down selectors not being the same size on high-DPI which was mentioned the other day.
